src/m365/viva/commands/engage/engage-community-add.ts

```diff
@@ -0,0 +1,220 @@
+        }
+
+        if (args.options.adminEntraIds && args.options.adminEntraUserNames) {
+          return `You can only specify either 'adminEntraIds' or 'adminEntraUserNames'`;
+        }
+
+        if (args.options.adminEntraIds) {
+          const adminEntraIds = args.options.adminEntraIds.split(',');
+          for (let i: number = 0; i < adminEntraIds.length; i++) {
+            const trimmedId: string = adminEntraIds[i].trim();
+            if (!validation.isValidGuid(trimmedId)) {
+              return `${trimmedId} is not a valid GUID`;
+            }
+          }
+        }
+
+        if (args.options.adminEntraUserNames) {
+          const adminEntraUserNames = args.options.adminEntraUserNames.split(',');
+          for (let i: number = 0; i < adminEntraUserNames.length; i++) {
+            const trimmedUPN: string = adminEntraUserNames[i].trim();
+            if (!validation.isValidUserPrincipalName(trimmedUPN)) {
+              return `${trimmedUPN} is not a valid UPN`;
+            }
+          }
+        }
+
+        return true;
+      }
+    );
+  }
+
+  #initTypes(): void {
+    this.types.string.push('displayName', 'description', 'privacy', 'adminEntraIds', 'adminEntraUserNames');
+  }
+
+  public async commandAction(logger: Logger, args: CommandArgs): Promise<void> {
+    const { displayName, description, privacy, adminEntraIds, adminEntraUserNames, wait } = args.options;
+
+    if (this.verbose) {
+      logger.logToStderr(`Creating a Viva Engage community with display name '${displayName}'...`);
+    }
+
+    try {
+      const requestOptions: CliRequestOptions = {
+        url: `${this.resource}/beta/employeeExperience/communities`,
+        headers: {
+          accept: 'application/json;odata.metadata=none',
+          'content-type': 'application/json;odata=nometadata'
+        },
+        responseType: 'json',
+        fullResponse: true,
+        data: {
+          displayName: displayName,
+          description: description,
+          privacy: privacy
+        }
+      };
+
+      let entraIds: string[] = [];
+
+      if (adminEntraIds) {
+        entraIds = adminEntraIds.split(',').map(id => this.getUserIdUrl(id));
+      }
+      else if (adminEntraUserNames) {
+        const userUPNs = adminEntraUserNames.split(',').map(upn => this.getUserId(upn));
+        const userIds = await Promise.all(userUPNs);
+        entraIds = userIds.map(id => this.getUserIdUrl(id));
+      }
+
+      if (entraIds.length > 0) {
+        requestOptions.data['owners@odata.bind'] = entraIds;
+      }
+
+      const res: any = await request.post(requestOptions);
+
+      const location: string = res.headers.location;
+
+      if (!wait) {
+        await logger.log(location);
+        return;
+      }
+
+      let status: string;
+
+      do {
+        if (this.verbose) {
+          logger.logToStderr(`Waiting 30 seconds...`);
+        }
+
+        await new Promise(resolve => setTimeout(resolve, 30000));
```

Comment:
I suggest that we use the `setTimeOut` from the `timers/promises` package. We have a few other commands that uses this as well.